wolvic/browser/autocomplete/wolvic_autofill_client.cc

```diff
@@ -0,0 +1,410 @@
+#include "components/security_state/core/security_state.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/ssl_status.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/browser/web_contents.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "ui/android/window_android.h"
+#include "url/origin.h"
+#include "wolvic/wolvic_browser_context.h"
+#include "wolvic/wolvic_content_browser_client.h"
+#include "wolvic/browser/autocomplete/wolvic_password_form_util.h"
+#include "wolvic/jni_headers/AutofillManager_jni.h"
+
+using base::android::AttachCurrentThread;
+
+namespace wolvic {
+
+// static
+void WolvicAutofillClient::CreateForWebContents(
+    content::WebContents* web_contents) {
+  DCHECK(web_contents);
+  if (!FromWebContents(web_contents)) {
+    web_contents->SetUserData(
+        UserDataKey(),
+        base::WrapUnique(new WolvicAutofillClient(web_contents)));
+  }
+}
+
+WolvicAutofillClient::~WolvicAutofillClient() = default;
+
+bool WolvicAutofillClient::IsOffTheRecord() {
+  return web_contents()->GetBrowserContext()->IsOffTheRecord();
+}
+
+scoped_refptr<network::SharedURLLoaderFactory>
+WolvicAutofillClient::GetURLLoaderFactory() {
+  return web_contents()
+      ->GetBrowserContext()
+      ->GetDefaultStoragePartition()
+      ->GetURLLoaderFactoryForBrowserProcess();
+}
+
+autofill::AutofillDownloadManager* WolvicAutofillClient::GetDownloadManager() {
+  if (!download_manager_) {
+    // Lazy initialization to avoid virtual function calls in the constructor.
+    download_manager_ =
+        std::make_unique<autofill::AutofillDownloadManager>(
+            this, GetChannel(), GetLogManager());
+  }
+  return download_manager_.get();
+}
+
+
+autofill::PersonalDataManager* WolvicAutofillClient::GetPersonalDataManager() {
+  return nullptr;
+}
+
+autofill::AutocompleteHistoryManager*
+WolvicAutofillClient::GetAutocompleteHistoryManager() {
+  return content::WolvicBrowserContext::FromWebContents(*web_contents())
+      ->GetAutocompleteHistoryManager();
+}
+
+PrefService* WolvicAutofillClient::GetPrefs() {
+  return content::WolvicBrowserContext::FromWebContents(*web_contents())
+      ->GetPrefService();
+}
+
+const PrefService* WolvicAutofillClient::GetPrefs() const {
+  return content::WolvicBrowserContext::FromWebContents(*web_contents())
+      ->GetPrefService();
+}
+
+syncer::SyncService* WolvicAutofillClient::GetSyncService() {
+  return nullptr;
+}
+
+signin::IdentityManager* WolvicAutofillClient::GetIdentityManager() {
+  return nullptr;
+}
+
+autofill::FormDataImporter* WolvicAutofillClient::GetFormDataImporter() {
+  return nullptr;
+}
+
+autofill::payments::PaymentsClient* WolvicAutofillClient::GetPaymentsClient() {
+  return nullptr;
+}
+
+autofill::StrikeDatabase* WolvicAutofillClient::GetStrikeDatabase() {
+  return nullptr;
+}
```

Comment:
What about adding all those `return nullptr` in the header file? Like `GetX() override { return nullptr; }` to simplify a bit this file? Same for the other similar getters including the ones with no code inside.

Reply:
We will meet the build error since Chromium don't allow to have the implementation of overridden method in header file.